src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -180,7 +182,7 @@ public static ITypeSymbol RemoveNullableIfPresent(this ITypeSymbol symbol)
                         interfaceMember,
                         solution,
                         typeSymbolCompilation,
```

Comment:
so, the input is basically, current symbol (actual implementation) and interface symbol (target interface symbol to match) ? and compilation of each symbol belong to?

wouldn't it be much cleaner if SymbolAndProjectId keep passed down and OriginalSymbolsMatch become Async method?

Reply:
by the way, when typeSymbolCompilation is null, VerifyForwardType seems doing this
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FindSymbols/SymbolFinder_Hierarchy.cs,508
to handle that case?

you might do the same for symbolToMatch?

I can see case where typeSymbolCompilation is null case?
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs,202

Reply:
> wouldn't it be much cleaner if SymbolAndProjectId keep passed down and OriginalSymbolsMatch become Async method? 

Maybe? At this time it'd be a lot of refactoring. I don't feel strongly enough that it would be cleaner, but if it would it should be a separate change.